base-lite/ub/ub.go

```diff
@@ -0,0 +1,313 @@
+	case "executable":
+		info, err := os.Stat(filePath)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error checking executable status of file %s: %s", filePath, err)
+			return false
+		}
+		return info.Mode()&0111 != 0 //check whether file is executable by anyone, use 0100 to check for execution rights for owner
+	case "existence":
+		if _, err := os.Stat(filePath); os.IsNotExist(err) {
+			return false
+		}
+		return true
+	case "writable":
+		return unix.Access(filePath, unix.W_OK) == nil
+	default:
+		fmt.Fprintf(os.Stderr, "Unknown operation %s", operation)
+	}
+	return false
+}
+
+func renderTemplate(templateFilePath string) bool {
+	templateFile, err := os.Open(templateFilePath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error opening file at path %s: %s", templateFilePath, err)
+		return false
+	}
+	bytes, err := io.ReadAll(templateFile)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error reading from file at path %s: %s", templateFilePath, err)
+		return false
+	}
+	funcs := template.FuncMap{
+		"getEnv":             getEnvOrDefault,
+		"splitToMapDefaults": splitToMapDefaults,
+	}
+	t, err := template.New("tmpl").Funcs(funcs).Parse(string(bytes))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error  %s: %s", templateFilePath, err)
+		return false
+	}
+	return buildTemplate(os.Stdout, *t)
+}
+
+func buildTemplate(writer io.Writer, template template.Template) bool {
+	err := template.Execute(writer, GetEnvironment())
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error building template file : %s", err)
+		return false
+	}
+	return true
+}
+
+func renderConfig(writer io.Writer, configSpec ConfigSpec) bool {
+	return writeConfig(writer, buildProperties(configSpec, GetEnvironment()))
+}
+
+// ConvertKey Converts an environment variable name to a property-name according to the following rules:
+// - a single underscore (_) is replaced with a .
+// - a double underscore (__) is replaced with a single underscore
+// - a triple underscore (___) is replaced with a dash
+// Moreover, the whole string is converted to lower-case.
+// The behavior of sequences of four or more underscores is undefined.
+func ConvertKey(key string) string {
+	re := regexp.MustCompile("[^_]_[^_]")
+	singleReplaced := re.ReplaceAllStringFunc(key, replaceUnderscores)
+	singleTripleReplaced := strings.ReplaceAll(singleReplaced, "___", "-")
+	return strings.ToLower(strings.ReplaceAll(singleTripleReplaced, "__", "_"))
+}
+
+// replaceUnderscores replaces every underscore '_' by a dot '.'
+func replaceUnderscores(s string) string {
+	return strings.ReplaceAll(s, "_", ".")
+}
+
+type ConfigSpec struct {
+	Prefixes          map[string]bool   `json:"prefixes"`
+	Excludes          []string          `json:"excludes"`
+	Renamed           map[string]string `json:"renamed"`
+	Defaults          map[string]string `json:"defaults"`
+	ExcludeWithPrefix string            `json:"excludeWithPrefix"`
+}
+
+// Contains returns true if slice contains element, and false otherwise.
```

Comment:
Use the https://pkg.go.dev/golang.org/x/exp/slices package instead (eventually this will be stdlib most likely)